### DISCUSSION TITLE
Bulk import of Prefixes

### DISCUSSION THREAD

**Discussion by mtaplin** [netbox-community/netbox#13261] [Help Wanted!] [answered]
Hi,
As a new user to Netbox, I'm trying without success to bulk import prefixes from a csv format.  I'am able to import all data with the exception of dhcp servers with constant fails with the following error:-

"Row 1 cf_dhcp_servers: Select a valid choice. 10.48.1.2 is not one of the available choices."

My import line looks like this:-

prefix,status,cf_add_to_dnac,cf_dhcp_servers
10.5.14.96/27,active,true,10.48.1.2

Ideally I'd like to expand this to be able to import 2 dhcp servers, something like 👎 

prefix,status,cf_add_to_dnac,cf_dhcp_servers
10.57.124.96/27,active,true,"10.48.1.2,10.48.1.3"

I've checked my choices and have selected many combinations to allow this import, all without success.
Any support you're able to offer would much appreciated.

Many thanks in advance

M Taplin

**Comment by mtinberg** [comment]
What does your validation rules and config for the custom field "dhcp_servers" look like?  One thing I notice from your CSV import is the leading space character before the field value " 10.48.1.2"​ which may not meet your validation rules.  Is this field an object reference to the ipam.ip-address data, selection field or a plain string field?


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: mtaplin ***@***.***>
Sent: Tuesday, July 25, 2023 8:35 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Bulk import of Prefixes (Discussion #13261)


Hi,
As a new user to Netbox, I'm trying without success to bulk import prefixes from a csv format. I'am able to import all data with the exception of dhcp servers with constant fails with the following error:-

"Row 1 cf_dhcp_servers: Select a valid choice. 10.48.1.2 is not one of the available choices."

My import line looks like this:-

prefix,status,cf_add_to_dnac,cf_dhcp_servers
10.5.14.96/27,active,true,10.48.1.2

Ideally I'd like to expand this to be able to import 2 dhcp servers, something like 👎

prefix,status,cf_add_to_dnac,cf_dhcp_servers
10.57.124.96/27,active,true,"10.48.1.2,10.48.1.3"

I've checked my choices and have selected many combinations to allow this import, all without success.
Any support you're able to offer would much appreciated.

Many thanks in advance

M Taplin

**Comment by mtaplin** [comment]
Hi Mark,

Many thanks for your very fast response.

Here is a screenshot of the validation for this custom field.

![Netbox_Validate](https://github.com/netbox-community/netbox/assets/140520272/16bacc09-0c96-4e27-a7fd-c9f2f5a281dd)

Regards,

Mark Taplin

**Reply by candlerb** [reply]
What Netbox version? The screen layout in Netbox 3.5.6 looks quite different to what you show (e.g. it says "Content Types" at the top, instead of "Assigned Models" in the middle)

Check in the "Choices" field that you don't have any spaces at the front, after the comma, or at the end.

With Netbox 3.5.6 it works fine for me. I created a custom field called "foobar" with type multiple selection on IPAM > Prefix, allowed values "1.1.1.1,2.2.2.2" and then imported the following:

```
prefix,status,cf_foobar
1.2.3.0/24,active,1.1.1.1
1.2.4.0/24,active,"1.1.1.1,2.2.2.2"
```

**Comment by mtaplin** [comment]
Hi candlerb,

Agreed my version is a little older (3.0.11) however I would still expect the import to work?

Can't see any spaces in my data.

Regards,

Mark Taplin

**Reply by candlerb** [reply, marked as answer]
> Agreed my version is a little older (3.0.11) however I would still expect the import to work?

It's 20 months old, and nothing is 100% perfect.

You won't be able to raise a bug report against an old version, because they are not maintained retrospectively. That is, there are no v3.0.x patch releases after v3.1 has been released.

You can check in the release notes to see if there's a comment about it being fixed in a particular version. I found the following in the release notes for [v3.1.6](https://github.com/netbox-community/netbox/releases/v3.1.6):

* [#8317](https://github.com/netbox-community/netbox/issues/8317) - Fix CSV import of multi-select custom field values

So I believe that's your answer: upgrade to 3.1.6 or later. If you're going to do that, you should at least go to the last release on v3.1, which is v3.1.11.

For every main version release, Netbox changes in incompatible ways, so you need to read the release notes carefully, in this case for v3.1.0.  If you want to upgrade all the way to v3.5.6, then make sure you read the release notes for v3.2.0, v3.3.0, v3.4.0 and v3.5.0 as well.  You may need to adjust integrations, custom scripts etc.

**Reply by mtaplin** [reply]
candlerb,
That's fantastic!  Many thanks for your support.

I'll head down the upgrade path!

Regards,

Mark Taplin
